girder/models/folder.py

```diff
@@ -691,19 +691,31 @@ def fileList(self, doc, user=None, path='', includeMetadata=False,
         if subpath:
             path = os.path.join(path, doc['name'])
         metadataFile = 'girder-folder-metadata.json'
-        for sub in self.childFolders(parentType='folder', parent=doc, user=user):
+
+        # Eagerly evaluate this list, as the MongoDB cursor can time out on long requests
+        childFolders = list(self.childFolders(
+            parentType='folder', parent=doc, user=user,
+            fields={'name': True, 'meta': includeMetadata}
```

Comment:
I don't think this will work.

> With the exception of the _id field, you may not have both inclusions and exclusions in the same projection document.

Reply:
Oops, you're right.